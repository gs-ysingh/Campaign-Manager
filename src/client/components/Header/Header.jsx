import React from "react";
import _ from 'lodash';
import { connect } from 'react-redux';
import create from '../../libs/create';
import styles from './Header.module.css';


export const Header = (props) => {
  return (
    <div className={styles.heading}>
      <div className={styles.label}>All Campaigns</div>
      <div className={styles.list}>
        <span>Campaign List</span>
        <button onClick={() => {
          const name = prompt("Please enter Campaign name");
          if(name) {
            const len = _.keys(props.data).length;
            const data = {};
            data[len + 1] = {
              name: name,
              createdAt: Date.now(),
              createdBy: props.userName,
              isRunning: false,
              userActions: [
                {
                  name: 'Created',
                  by: props.userName,
                  label: 'Campaign Created',
                }
              ]
            };
            props.create(data);
          }
        }}>+ Create New</button>
      </div>
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    data: state.data,
    activeIndex: state.activeIndex,
    userName: state.userName
  };
}

export const mapDispatchToProps = dispatch => ({
  create(obj) {
    dispatch(create(obj));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);