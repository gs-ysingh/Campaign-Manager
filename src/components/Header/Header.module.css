.label {
  height: 40px;
  background: #f7fbff;
  padding-left: 40px;
  display: flex;
  align-items: center;
}

.list {
  height: 80px;
  background: #fff;
  display: flex;
  align-items: center;
  padding-left: 40px;
}

button {
  font-size: 12px;
  height: 30px;
  width: 100px;
  border: 0;
  background: #4a8ee0;
  color: #FFF;
  margin-left: 20px;
  cursor: pointer;
}