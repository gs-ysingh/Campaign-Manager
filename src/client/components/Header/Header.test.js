import React from 'react';
import { create } from 'react-test-renderer';
import { Header } from './Header';

test('test snapshot', () => {
  const mockCreate = jest.fn();
  const c = create(<Header create={mockCreate} />);
  expect(c.toJSON()).toMatchSnapshot();
});