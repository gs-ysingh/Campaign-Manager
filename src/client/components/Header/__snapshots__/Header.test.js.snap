// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div>
  <div>
    All Campaigns
  </div>
  <div>
    <span>
      Campaign List
    </span>
    <button
      onClick={[Function]}
    >
      + Create New
    </button>
  </div>
</div>
`;
